### PR TITLE
database endpoint revision

### PR DESCRIPTION
**Pull request by Hong-Revo**
Handle changed token management in AzureRMR; endpoint property parsing

### CODE REVIEW

**Comment by Hong-Revo** [comment]
Pending #7

File: tests/testthat/test01_database.R [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+rg <- AzureRMR::az_rm$
+        new(tenant=tenant, app=app, password=password)$
+        get_subscription(subscription)$
+        get_resource_group(rgname)
+
+
+
+test_that("Database resource functions work",
+{
+    srv <- rg$get_kusto_cluster(srvname))
```
Comment (alexkyllo) [inline tests/testthat/test01_database.R]: Several of your tests have statements with extra closing parens.

File: tests/testthat/test00_resource.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    skip("Authentication tests skipped: ARM credentials not set")
+
+
+sub <- AzureRMR::az_rm$new(tenant=tenant, app=app, password=password)$get_subscription(subscription)
+
+Sys.setenv(AZ_TEST_RG=paste(sample(letters, 20, replace=TRUE), collapse=""))
+Sys.setenv(AZ_TEST_KUSTO=paste(sample(letters, 10, replace=TRUE), collapse=""))
+
+
+test_that("ARM interface works",
```
Comment (alexkyllo) [inline tests/testthat/test00_resource.R]: This test fails with:

```
test00_resource.R:26: error: ARM interface works
Not Found (HTTP 404). Failed to complete Resource Manager operation. Message:
The resource with identifier
'/subscriptions/35975484-5360-4e67-bf76-14fcb0ab5b9d/resourceGroups/plutgdjtuycnpqtwtwmd/providers/Microsoft.Kusto/Clusters/lkgznvvwxa'
is not found.
1: rg$create_kusto_cluster(srvname, wait = TRUE) at /home/alex/projects/AzureKusto/tests/testthat/test00_resource.R:26
2: az_kusto$new(self$token, self$subscription, self$name, type = "Microsoft.Kusto/clusters", 
       name = name, location = location, sku = list(name = node_size, tier = "Standard"), 
       ..., wait = wait) at /home/alex/projects/AzureKusto/R/add_methods.R:8
3: .subset2(public_bind_env, "initialize")(...)
4: private$init_and_deploy(..., wait = wait)
5: private$res_op()
6: call_azure_rm(self$token, self$subscription, op, ..., api_version = api_version)
7: call_azure_url(token, httr::build_url(url), ...)
8: process_response(res, match.arg(http_status_handler))
9: handler(response, paste0("complete Resource Manager operation. Message:\n", sub("\\.$", 
       "", arm_error_message(response))))
```

File: tests/testthat/test00_resource.R [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    expect_true(sub$resource_group_exists(rgname))
+
+    srvname <- Sys.getenv("AZ_TEST_KUSTO")
+    srv <- rg$create_kusto_cluster(srvname, wait=TRUE)
+
+    expect_true(is_kusto_cluster(srv)))
+})
+
+
+test_that("Cluster resource functions work",
```
Comment (alexkyllo) [inline tests/testthat/test00_resource.R]: This test fails with:

```
test00_resource.R:34: error: Cluster resource functions work
object 'rg' not found
1: .handleSimpleError(function (e) 
   {
       handled <<- TRUE
       test_error <<- e
       options(expressions = expressions_opt_new)
       on.exit(options(expressions = expressions_opt), add = TRUE)
       e$expectation_calls <- frame_calls(11, 2)
       test_error <<- e
       register_expectation(e)
       e$handled <- TRUE
       test_error <<- e
   }, "object 'rg' not found", quote(eval(code, test_env))) at /home/alex/projects/AzureKusto/tests/testthat/test00_resource.R:34
2: eval(code, test_env)
```

File: tests/testthat/test00_resource.R [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    tok <- srv$get_aad_token(tenant)
+    expect_true(AzureRMR::is_azure_token(tok) && tok$tenant == tenant)
+
+    conf_dir <- rappdirs::user_config_dir(appname="AzureKusto", appauthor="AzureR", roaming=FALSE)
+    expect_true(length(dir(conf_dir) >= 1))
+})
+
+
+test_that("AAD token functions work",
```
Comment (alexkyllo) [inline tests/testthat/test00_resource.R]: This test fails with:
```
test00_resource.R:52: error: AAD token functions work
'normalize_tenant' is not an exported object from 'namespace:AzureRMR'
1: get_kusto_token(cluster = srvname, location = "australiaeast", tenant = tenant) at /home/alex/projects/AzureKusto/tests/testthat/test00_resource.R:52
2: AzureRMR::normalize_tenant at /home/alex/projects/AzureKusto/R/cluster.R:19
3: getExportedValue(pkg, name)
4: stop(gettextf("'%s' is not an exported object from 'namespace:%s'", name, getNamespaceName(ns)), 
       call. = FALSE, domain = NA)
```

Comment (alexkyllo) [reply tests/testthat/test00_resource.R]: I'll install the latest AzureRMR and see if that fixes this error

Comment (alexkyllo) [reply tests/testthat/test00_resource.R]: Different error after reinstalling AzureRMR:

```
test00_resource.R:52: error: AAD token functions work
Bad Request (HTTP 400).
1: get_kusto_token(cluster = srvname, location = "australiaeast", tenant = tenant) at /home/alex/projects/AzureKusto/tests/testthat/test00_resource.R:52
2: AzureRMR::get_azure_token(resource = server, tenant = tenant, app = .kusto_app_id, 
       auth_type = "device_code", aad_host = "https://login.microsoftonline.com/") at /home/alex/projects/AzureKusto/R/cluster.R:24
3: auth_with_device(base_url, app, resource_host)
4: AzureToken$new(endp, app, user_params = list(resource = resource), use_device = TRUE, 
       client_credentials = FALSE)
5: .subset2(public_bind_env, "initialize")(...)
6: private$init_with_device(user_params)
7: httr::stop_for_status(res)
```

File: tests/testthat/test00_resource.R (hunk 2 above)
Comment (alexkyllo) [reply tests/testthat/test00_resource.R]: This passes after reinstalling dev AzureRMR

**Comment by Hong-Revo** [comment]
Yeah, the tests aren't done yet. When the tests are done there'll be a block to clean everything up, right now there's artifacts left behind.

Also, you should be substituting your own credentials in the test_creds.R file. Right now everything is being created in my sub.

**Comment by alexkyllo** [comment]
Ok, just let me know when you're ready for a review, meanwhile I'll go get the credentials for my own Azure sub.

**Comment by Hong-Revo** [comment]
Tests should be passing now

**Comment by alexkyllo** [comment]
Ok cool, I'll pull your branch and make sure that I can get the tests to pass on my box. Then, since you've added integration tests in this PR, what do you think about setting up TravisCI to test future PRs? It looks like you've done so for AzureRMR already.

**Comment by alexkyllo** [comment]
On second thought, I don't think I want to keep a Kusto cluster running in my subscription all the time just so that I can run the integration tests locally. I'll just skip them for now and take your word for it that they're passing.

**Pull request merged**